File: src/AoC2020/Day1/Day1.java
package AoC2020.Day1;

import utils.FileUtil;
import utils.ExecutionTime;
import java.util.List;

public class Day1 {

    private static final  List<Integer> input = FileUtil.readFileInt("src/AoC2020/Day1/day1.txt");

    public static void main(String[] args) {
        ExecutionTime.start();
        part1();
        part2();
        ExecutionTime.stop();
    }

    public static void part1(){
        for (Integer a : input) {
            for (Integer b : input) {
                if (a + b == 2020) {
                    System.out.println("Part 1: " + a * b);
                    return;
                }
            }
        }
    }

    public static void part2(){
        for (Integer a : input) {
            for (Integer b : input) {
                for (Integer c : input) {
                    if (a + b + c == 2020) {
                        System.out.println("Part 2: " + a * b * c);
                        return;
                    }
                }
            }
        }
    }
}

File: src/AoC2020/Day1/Day1.kt
package AoC2020.Day1

import java.io.File
import utils.ExecutionTime

val input = File("src/AoC2020/Day1/day1.txt").readLines().map { it.toInt() }

fun part1(): Unit = run outForeach@{ input.forEach { k -> input.drop(0).forEach { e -> if (k+e == 2020)  {return@outForeach println(k*e); } } } }

fun part2(): Unit = run outForeach@{ input.forEach { i -> input.drop(0).forEach { k -> input.drop(0).forEach { e -> if (i+k+e == 2020)  {return@outForeach println(i*k*e); } } } } }

fun main() {
    ExecutionTime.start()
    //part1()
    part2()
    ExecutionTime.stop()


}